support/signal.c

```diff
@@ -379,7 +487,7 @@ wait_for_any (signal_info** signals, int count, int *currfd, struct pollfd* fd_s
 					r = read (h->read_fd, &c, 1);
 				} while (keep_trying (r) && !shutting_down ());
 				if (idx == -1)
-					idx = i;
+					idx = i; // This is the first index we read (unless we're shutting down)
```

Comment:
Needless comment, and I'm not sure how accurate it is. What does "shutting down" have to do with the index?